### PR TITLE
Add prompt_generator module and identify_file_type util function

### PR DESCRIPTION
**Pull request by ka-sarthak**
Prompt generator module to handle the nomad schema and input raw files containing unstructured data.

### CODE REVIEW

**Comment by JosePizarro3** [comment]
I can review this later today, but did you check: https://python.langchain.com/docs/use_cases/extraction/how_to/handle_files/? 🙂

**Comment by JosePizarro3** [comment]
This is also interesting for @ladinesa; I think he can help you much better than me on this

**Comment by ka-sarthak** [comment]
Thanks for the link to LangChain. Indeed we can make use of their readers to handle the raw data files.

I added @ladinesa for review.

File: src/llmad/utils.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import warnings
+
+
+def identify_file_type(file_path: str) -> str:
```
Comment (ladinesa) [inline src/llmad/utils.py:4]: we can use magic library for this

Comment (ka-sarthak) [reply src/llmad/utils.py]: Indeed, I have now implemented `indentify_mime_type` based on magic

**Pull request merged**